diff -up mail-notification-5.4/build/src/mn-evolution-folder-tree-server.c.libemail mail-notification-5.4/build/src/mn-evolution-folder-tree-server.c
--- mail-notification-5.4/build/src/mn-evolution-folder-tree-server.c.libemail	2012-02-08 10:18:21.261982020 +0100
+++ mail-notification-5.4/build/src/mn-evolution-folder-tree-server.c	2012-02-08 10:19:27.879757995 +0100
@@ -30,7 +30,11 @@
 #include <mail/mail-component.h>
 #endif
 #if EDS_CHECK_VERSION(3,1,0)
+#if EDS_CHECK_VERSION(3,3,5)
+#include <libemail-engine/e-mail-folder-utils.h>
+#else
 #include <mail/e-mail-folder-utils.h>
+#endif
 #include <mail/e-mail-browser.h>
 #include <mail/e-mail-reader.h>
 #include <shell/e-shell.h>
diff -up mail-notification-5.4/build/src/mn-evolution-folder-tree-server-private.h.libemail mail-notification-5.4/build/src/mn-evolution-folder-tree-server-private.h
--- mail-notification-5.4/build/src/mn-evolution-folder-tree-server-private.h.libemail	2012-02-08 10:18:32.773116469 +0100
+++ mail-notification-5.4/build/src/mn-evolution-folder-tree-server-private.h	2012-02-08 10:19:27.879757995 +0100
@@ -5,9 +5,13 @@
 
 #include "mn-evolution-folder-tree-server.h"
 #include <libedataserver/eds-version.h>
+#if EDS_CHECK_VERSION(3,3,5)
+#include <libemail-engine/e-mail-session.h>
+#else
 #if EDS_CHECK_VERSION(2,91,0)
 #include <mail/e-mail-session.h>
 #endif
+#endif
 
 #ifdef __cplusplus
 extern "C" {
diff -up mail-notification-5.4/build/src/mn-evolution-server.c.libemail mail-notification-5.4/build/src/mn-evolution-server.c
--- mail-notification-5.4/build/src/mn-evolution-server.c.libemail	2012-02-08 10:18:43.965246700 +0100
+++ mail-notification-5.4/build/src/mn-evolution-server.c	2012-02-08 10:19:27.909758343 +0100
@@ -37,12 +37,17 @@
 #include <mail/em-message-browser.h>
 #endif
 #include <mail/em-utils.h>
+#if EDS_CHECK_VERSION(3,3,5)
+#include <libemail-engine/e-mail-session.h>
+#include <libemail-engine/mail-tools.h>
+#else
+#include <mail/mail-tools.h>
 #if EDS_CHECK_VERSION(2,91,0)
 #include <mail/e-mail-session.h>
 #else
 #include <mail/mail-session.h>
 #endif
-#include <mail/mail-tools.h>
+#endif
 #include "mn-evolution.h"
 #include "mn-evolution-folder-tree-server.h"
 #include "mn-evolution-plugin.h"
diff -up mail-notification-5.4/jbsrc/jb.c.libemail mail-notification-5.4/jbsrc/jb.c
--- mail-notification-5.4/jbsrc/jb.c.libemail	2012-02-08 10:18:55.885385460 +0100
+++ mail-notification-5.4/jbsrc/jb.c	2012-02-08 10:19:27.940758702 +0100
@@ -635,6 +635,7 @@ jb_package_add_resources (void)
       jb_group_add_data_file(group, "org-jylefort-mail-notification.eplug", "$evolution-plugin-dir");
 
       jb_compile_options_add_cflags(object->compile_options, "$evolution-plugin-cflags");
+      jb_compile_options_add_cflags(object->compile_options, "$libemail-engine-cflags");
 
       jb_group_add_dbus_interface(group,
 				  "org.freedesktop.DBus.Properties",
@@ -662,6 +663,7 @@ jb_package_add_resources (void)
 
       jb_compile_options_add_package(plugin->compile_options, "gettext");
       jb_compile_options_add_package(plugin->compile_options, "evolution-plugin");
+      jb_compile_options_add_package(plugin->compile_options, "libemail-engine");
       jb_compile_options_add_package(plugin->compile_options, "dbus");
 
       jb_group_add_dbus_interface(group,
diff -up mail-notification-5.4/jbsrc/lib/src/extras/jb-evolution-plugin.c.libemail mail-notification-5.4/jbsrc/lib/src/extras/jb-evolution-plugin.c
--- mail-notification-5.4/jbsrc/lib/src/extras/jb-evolution-plugin.c.libemail	2012-02-08 10:19:06.006503405 +0100
+++ mail-notification-5.4/jbsrc/lib/src/extras/jb-evolution-plugin.c	2012-02-08 10:19:27.969759042 +0100
@@ -63,5 +63,5 @@ jb_evolution_plugin_check (const char *m
       g_free(plugindir);
     }
 
-  return TRUE;
+  return jb_check_packages("Evolution", "libemail-engine", "libemail-engine");
 }
diff -up mail-notification-5.4/src/mn-evolution-plugin.c.libemail mail-notification-5.4/src/mn-evolution-plugin.c
--- mail-notification-5.4/src/mn-evolution-plugin.c.libemail	2012-02-08 10:19:16.006619648 +0100
+++ mail-notification-5.4/src/mn-evolution-plugin.c	2012-02-08 10:19:27.970759054 +0100
@@ -27,8 +27,13 @@
 #include <camel/camel.h>
 #include <libedataserver/eds-version.h>
 #include <mail/em-event.h>
+#if EDS_CHECK_VERSION(3,3,5)
+#include <libemail-engine/e-mail-folder-utils.h>
+#include <libemail-engine/mail-tools.h>
+#else
 #include <mail/mail-tools.h>
 #include <mail/e-mail-folder-utils.h>
+#endif
 #include "mn-evolution.h"
 #include "mn-evolution-server.h"
 #include "mn-evolution-plugin.h"
