diff -aurN mail-notification-3.0.orig/data/mail-notification.desktop.in mail-notification-3.0/data/mail-notification.desktop.in
--- mail-notification-3.0.orig/data/mail-notification.desktop.in	2006-05-03 03:55:40.000000000 +0200
+++ mail-notification-3.0/data/mail-notification.desktop.in	2006-06-14 18:38:55.441261912 +0200
@@ -4,7 +4,7 @@
 Icon=mail-notification.png
 _Comment=Get notified when new mail arrives
 Type=Application
-Categories=GNOME;Application;Network;
+Categories==GTK;GNOME;Network;Email;
 Exec=mail-notification --sm-disable
 Terminal=false
 StartupNotify=true
diff -aurN mail-notification-3.0.orig/data/mail-notification-properties.desktop.in mail-notification-3.0/data/mail-notification-properties.desktop.in
--- mail-notification-3.0.orig/data/mail-notification-properties.desktop.in	2006-03-04 06:04:49.000000000 +0100
+++ mail-notification-3.0/data/mail-notification-properties.desktop.in	2006-06-14 18:39:16.072125544 +0200
@@ -4,7 +4,7 @@
 Icon=mail-notification.png
 _Comment=Configure Mail Notification
 Type=Application
-Categories=GNOME;Application;Settings;
+Categories=GTK;GNOME;Settings;
 Exec=mail-notification --display-properties
 Terminal=false
 StartupNotify=true
