--- mail-notification-5.0/data/mail-notification.desktop.in.orig	2008-03-01 11:02:29.000000000 +0100
+++ mail-notification-5.0/data/mail-notification.desktop.in	2008-03-01 11:02:48.000000000 +0100
@@ -4,8 +4,8 @@
 Icon=mail-notification
 _Comment=Get notified when new mail arrives
 Type=Application
-Categories=GNOME;GTK;Network;
+Categories=GNOME;GTK;Network;Email
 Exec=mail-notification --sm-disable
 Terminal=false
 StartupNotify=true
-X-GNOME-DocPath=mail-notification/mail-notification.xml
\ No newline at end of file
+X-GNOME-DocPath=mail-notification/mail-notification.xml
