diff -urN mail-notification-0.99.orig/data/mail-notification.desktop.in mail-notification-0.99/data/mail-notification.desktop.in
--- mail-notification-0.99.orig/data/mail-notification.desktop.in	2004-11-17 15:18:57.647008664 +0100
+++ mail-notification-0.99/data/mail-notification.desktop.in	2004-11-17 15:20:01.779259080 +0100
@@ -4,6 +4,6 @@
 Icon=mail-notification.png
 _Comment=Get notified when new mail arrives
 Type=Application
-Categories=GNOME;Application;Network;
+Categories=GTK;GNOME;Network;Email;
 Exec=mail-notification
 Terminal=false
