--- mail-notification-4.1/configure.ac~	2007-06-26 13:39:38.000000000 +0200
+++ mail-notification-4.1/configure.ac	2007-10-02 19:41:25.833527486 +0200
@@ -152,20 +152,24 @@
 
 if MN_FEATURE_ENABLED(evolution); then
 	evolution=no
-	AM_PATH_EVOLUTION_PLUGIN(2.12,, [evolution=yes])
-	if test $evolution = no; then
-		AM_PATH_EVOLUTION_PLUGIN(2.10,, [evolution=yes])
-	fi
-	if test $evolution = no; then
-		AM_PATH_EVOLUTION_PLUGIN(2.8,, [evolution=yes])
-	fi
-	if test $evolution = no; then
-		AM_PATH_EVOLUTION_PLUGIN(2.6,, [evolution=yes])
-	fi
-	if test $evolution = no; then
-		AM_PATH_EVOLUTION_PLUGIN(2.4,, [evolution=yes])
-	fi
-	if test $evolution = no; then
+	PKG_CHECK_MODULES(EVOLUTION_PLUGIN, [evolution-plugin], [found=yes], [found=no])
+
+	if test $found = yes; then
+		AC_MSG_CHECKING([for the Evolution plugin directory])
+		evolution_plugindir=`$PKG_CONFIG --variable=plugindir evolution-plugin 2>/dev/null`
+		if test -n "$evolution_plugindir"; then
+			AC_MSG_RESULT([$evolution_plugindir])
+		else
+			AC_MSG_RESULT([not found])
+			found=no
+		fi
+	fi
+
+	if test $found = yes; then
+		evolution=yes
+		AC_SUBST(evolution_plugindir)
+	else
+		evolution_plugindir=""
 		MN_FEATURE_DISABLE(evolution, [Evolution not found])
 	fi
 fi
diff -Nur mail-notification-4.1.dfsg.1/src/mn-evolution-folder-tree-control.c mail-notification-4.1.dfsg.1.new/src/mn-evolution-folder-tree-control.c
--- mail-notification-4.1.dfsg.1/src/mn-evolution-folder-tree-control.c	2007-06-26 13:36:57.000000000 +0200
+++ mail-notification-4.1.dfsg.1.new/src/mn-evolution-folder-tree-control.c	2007-08-15 16:54:28.000000000 +0200
@@ -126,7 +126,7 @@
     switch (arg_id)
       {
       case PROP_SELECTED_URI:
-	em_folder_tree_set_selected(tree, BONOBO_ARG_GET_STRING(arg));
+	em_folder_tree_set_selected(tree, BONOBO_ARG_GET_STRING(arg), TRUE);
 	break;
 
       default:
